plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FactRetrieverContext } from '@backstage/plugin-tech-insights-node';
+import { CatalogClient } from '@backstage/catalog-client';
+import { RELATION_OWNED_BY } from '@backstage/catalog-model';
+
+const applicableEntityKinds = [
+  'component',
+  'api',
+  'template',
+  'domain',
+  'resource',
+  'system',
+];
+
+const entityKindFilter = { field: 'kind', values: applicableEntityKinds };
+
+export const entityFactRetriever = {
+  id: 'entityRetriever',
+  version: '0.0.1',
+  entityFilter: [
+    {
+      field: 'kind',
+      values: applicableEntityKinds,
+    },
+  ],
+  schema: {
+    hasSpecWithOwner: {
+      type: 'boolean',
+      description: 'The spec.owner field is set',
+    },
+    hasSpecWithGroupOwner: {
+      type: 'boolean',
+      description: 'The spec.owner field is set and refers to a group',
+    },
+    hasOwnedByRelation: {
+      type: 'boolean',
+      description: 'The entity has an owned_by relation',
+    },
+    hasOwnedByRelationWithGroupTarget: {
+      type: 'boolean',
+      description: 'The entity has an owned_by relation which targets a group',
+    },
+  },
+  handler: async ({ discovery }: FactRetrieverContext) => {
+    const catalogClient = new CatalogClient({
+      discoveryApi: discovery,
+    });
+    const entities = await catalogClient.getEntities({
```

Comment:
I wonder if it makes sense / was considered to move this logic to the engine and just pass the matching entities to the handler? given the entity filter is a field on the object anyway (I haven't considered this deeply just an idea)

Reply:
It might be better suited in there tbh. The case for not moving it there would be to enable these fact retrievers to get data for items that are not directly tied to the entity structure of Backstage which was one of the questions in the original PRFC. Though moving it there wouldn't prevent it tbh. 

Maybe the filter could be passed in as a context item to the retriever so the implementer can decide whether to use it or not?

Reply:
> Maybe the filter could be passed in as a context item to the retriever so the implementer can decide whether to use it or not?

That's better for sure. Then at least we know there's probably a link between that field and the actual entities polled and is just a bit neater .